src/G2/Equiv/Approximation.hs

```diff
@@ -0,0 +1,125 @@
+module G2.Equiv.Approximation where
+
+import G2.Equiv.Types
+import G2.Language
+import qualified G2.Language.Approximation as A
+import qualified G2.Language.ExprEnv as E
+
+import qualified Data.HashSet as HS
+import qualified Data.HashMap.Lazy as HM
+import Data.List
+import Data.Maybe
+
+moreRestrictive :: StateET
+                -> StateET
+                -> HS.HashSet Name
+                -> (HM.HashMap Id Expr, HS.HashSet (Expr, Expr))
+                -> Either [Lemma] (HM.HashMap Id Expr, HS.HashSet (Expr, Expr))
+moreRestrictive s1 s2 =
+    let
+        lkp n s = lookupConcOrSymBoth n (expr_env s) (opp_env $ track s)
+    in
+    A.moreRestrictive moreRestrictiveCont generateLemma lkp s1 s2
+
+moreRestrictiveCont :: StateET
+                    -> StateET
+                    -> HS.HashSet Name
+                    -> (HM.HashMap Id Expr, HS.HashSet (Expr, Expr))
+                    -> Bool -- ^ indicates whether this is part of the "active expression"
+                    -> [(Name, Expr)] -- ^ variables inlined previously on the LHS
+                    -> [(Name, Expr)] -- ^ variables inlined previously on the RHS
+                    -> Expr
+                    -> Expr
+                    -> Either [Lemma] (HM.HashMap Id Expr, HS.HashSet (Expr, Expr))
+moreRestrictiveCont s1 s2 ns hm active n1 n2 e1 e2 =
```

Comment:
What's the meaning of "Cont" here?

Reply:
"Continuation" although I guess that's not really what this is.  Not sure what a better name would be...

Reply:
Well, the continuation comes into play later in the helper functions that this uses, so "Cont" might be the best option for a name.

Reply:
I mean, it's close enough for me, if you're happy with it.  Especially since it's really just for use in moreRestrictive, as also defined in G2.Equiv.Continuation